File: src/main/resources/ng-frontend/src/app/category-view.ts
import {Category} from './category';
import {Topic} from './topic';
import {Pageable} from './pageable';
import {SortLink} from './sort-link';

export interface CategoryView {
  categoryView: {
    category: Category,
    tree: Category[],
    topics: Topic[],
    pageable: Pageable,
    pageCount: number,
  },
  sortLinks: SortLink[],
}

File: src/main/java/de/dafri/dwb/view/CategoryView.java
package de.dafri.dwb.view;

import org.springframework.data.domain.Pageable;

import java.util.List;

public record CategoryView(List<CategoryTreeViewItem> tree, CategoryTreeViewItem category, List<TopicListViewItem> topics,
                           Pageable pageable, int pageCount) {
}
